akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala

```diff
@@ -298,7 +298,9 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
       override def onPull() = if (isAvailable(in)) {
         sourceOut.push(grab(in))
       } else {
-        pull(in)
+        if (!hasBeenPulled(in)) {
```

Comment:
ok, some misunderstanding of the relation between `isAvailable` and `hasBeenPulled`. Scaladoc mention some invariants, but the opposite of them are probably not true.